nifi-nar-bundles/nifi-morphlines-bundle/nifi-morphlines-processors/src/main/java/org/apache/nifi/processors/morphlines/ExecuteMorphline.java

```diff
@@ -0,0 +1,253 @@
+        }
+        final MorphlineContext morphlineContext = new MorphlineContext.Builder().setSettings(settings).build();
+
+        final Collector collectorRecord = new Collector();
+        final Command morphline = new Compiler().compile(morphlinesFile, morphlinesId, morphlineContext, collectorRecord);
+
+        try{
+            flowFile = session.write(flowFile, new StreamCallback() {
+                @Override
+                public void process(InputStream in, OutputStream out) throws IOException {
+                    Record record = new Record();
+                    StreamUtils.fillBuffer(in, value);
+                    record.put(Fields.ATTACHMENT_BODY, value);
+                    Notifications.notifyStartSession(morphline);
+                    if (morphline.process(record)) {
+                        List<Record> results = collectorRecord.getRecords();
+                        for (Iterator<Record> it = results.iterator(); it.hasNext();) {
+                            Record result = it.next();
+                            if (result.getFirstValue(morphlinesOutputField) != null) {
+                                String outputValue = it.next().getFirstValue(morphlinesOutputField).toString() + "/n";
+                                out.write(outputValue.getBytes());
+                                written.incrementAndGet();
+                            } else {
+                                getLogger().warn(String.format("Unable to get %s within processed record: %s", morphlinesOutputField, result.toString()));
+                            }
+                        }
+                        Notifications.notifyCommitTransaction(morphline);
+                    }
+                }
+            });
+
+            if (written.get() > 0L) {
+                // false to only update if file transfer is successful
+                session.adjustCounter("Processed records in morphlines", written.get(), false);
+                session.transfer(flowFile, REL_SUCCESS);
+                session.transfer(originalFlowFile, REL_ORIGINAL);
+            } else {
+                getLogger().warn(String.format("Morphlines transformations did not march any of the input records for %s Morphlines ID", morphlinesId));
+                session.transfer(flowFile, REL_ORIGINAL);
+                session.transfer(originalFlowFile, REL_FAILURE);
+            }
+        } catch (ProcessException e) {
+            getLogger().error("Error while processing the flowFile through Morphlines");
+            session.transfer(flowFile, REL_ORIGINAL);
+            session.transfer(originalFlowFile, REL_FAILURE);
+            Notifications.notifyRollbackTransaction(morphline);
+            morphlineContext.getExceptionHandler().handleException(e, null);
+        }
+        Notifications.notifyShutdown(morphline);
+    }
+
+    private static final class Collector implements Command {

```

Comment:
The same as above, this implementation is not going to work with big input file

Reply:
Same answer as the above.